Rename handlers and events for entity processing consistency

lynconsl opened the pull request:
The change from `onClientPedsProcessed` to `onClientEntitiesProcessed` indicates that the event is not only related to pedestrians, but can also be used on various elements, for example: #3922 The idea to revive the event came from this issue.

The ped manipulators have also been renamed to entities.

Commit:
Rename handlers and events for entity processing consistency

botder commented:
Can you add a warning/error to addEventHandler whenever somebody tries to register a handler for `onClientPedsProcessed`?

ghost commented:
> Can you add a warning/error to addEventHandler whenever somebody tries to register a handler for `onClientPedsProcessed`?

I'll find a way to do it

tederis commented:
This event should be named "onClientPostUpdate". It will describe the meaning of the event better.

Commit:
Rename event from onClientEntitiesProcessed to onClientPostUpdate and update related references

botder commented on Client/mods/deathmatch/logic/CClientGame.cpp:
```diff
@@ -3872,11 +3872,17 @@ void CClientGame::PostWorldProcessHandler()
+    {
+        throw std::runtime_error("'onClientPedsProcessed' is no longer supported, renamed to 'onClientPostUpdate'");
+        return; 
+    }
```
The better solution would be to just assume `onClientPostUpdate` when somebody does `addEventHandler("onClientPedsProcessed", ...)` but also show a warning.

Commit:
Fix

botder commented on Client/mods/deathmatch/logic/CClientGame.cpp:
```diff
@@ -3872,11 +3872,17 @@ void CClientGame::PostWorldProcessHandler()
+    {
+        CStaticFunctionDefinitions::AddEventHandler("onClientPedsProcessed", "onClientPostUpdate", "The 'onClientPedsProcessed' event is deprecated and has been renamed to 'onClientPostUpdate'.");
+        return; 
+    }
```
I don't know how you came up with the idea to put that here, remove it and adapt `CStaticFunctionDefinitions::AddEventHandler`.

MohabCodeX committed:
Merge branch 'master' into rename-event

MohabCodeX committed:
Fix backward compatibility for onClientPedsProcessed event

- Move deprecation handling from event trigger to AddEventHandler
- Remove broken compatibility code from PostWorldProcessEntitiesAfterPreRenderHandler
- Add proper warning when scripts use deprecated event name
- Ensure seamless migration: old event name redirects to onClientPostUpdate

Commit:
Merge branch 'multitheftauto:master' into rename-event

Commit:
Merge branch 'master' into rename-event

MohabCodeX committed:
Merge branch 'rename-event' into rename-event

MohabCodeX committed:
Merge branch 'multitheftauto:master' into rename-event

Commit:
Merge pull request #1 from MohabCodeX/rename-event

Fix backward compatibility for onClientPedsProcessed event

Commit:
r & a

Commit:
Merge branch 'master' into rename-event
